src/centers.rs

```diff
@@ -54,4 +64,69 @@ impl Center {
         }
         centroids
     }
+
+    /// Calculates the geographical center of gravity for each cluster.
+    ///
+    /// This method is specifically designed for geographical data where each point
+    /// is represented by latitude and longitude coordinates.
+    ///
+    /// # Arguments
+    ///
+    /// * `data` - A slice of vectors, where each vector contains [latitude, longitude] in degrees.
+    /// * `labels` - A slice of cluster labels corresponding to each data point.
+    ///
+    /// # Returns
+    ///
+    /// A vector of cluster centers, where each center is a vector of [latitude, longitude] in degrees.
+    ///
+    /// # Notes
+    ///
+    /// - Assumes input coordinates are in degrees.
+    /// - Output coordinates are in degrees.
+    /// - Points with label -1 are considered noise and are ignored in calculations.
+    /// - Uses a spherical approximation of the Earth for calculations.
+    fn centers_of_gravity<T: Float + AddAssign>(
+        &self,
+        data: &[Vec<T>],
+        labels: &[i32],
+    ) -> Vec<Vec<T>> {
+        let n_clusters = labels
+            .iter()
+            .filter(|&&label| label != -1)
+            .collect::<HashSet<_>>()
+            .len();
+        let mut centers = vec![vec![T::zero(), T::zero()]; n_clusters];
+        let mut counts = vec![T::zero(); n_clusters];
+
+        for (point, &label) in data.iter().zip(labels.iter()) {
+            if label != -1 {
+                let cluster_index = label as usize;
+                centers[cluster_index][0] += point[0].to_radians();
+                centers[cluster_index][1] += point[1].to_radians();
+                counts[cluster_index] += T::one();
```

Comment:
I appreciate it will be a little more verbose, but can we do this without the additional `AddAssign` trait bound? Strictly speaking, per semantic versioning rules, this is a breaking change that would require a new major version release. 
I know that Rust's standard floating point types are already `AddAssign` but it is theoretically possible that there are other types that implement `Float` that do not implement `AddAssign`.

Reply:
Updated https://github.com/tom-whitehead/hdbscan/pull/26/commits/f3c74ff8cdcdcc6b43a7289d8bbf2a8316281ff2